spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisUtils.java

```diff
@@ -61,8 +62,14 @@ public static boolean isUnlinkAvailable(RedisOperations<?, ?> redisOperations) {
 			Properties info = redisOperations.execute(
 					(RedisCallback<Properties>) connection -> connection.serverCommands().info(SECTION));
 			if (info != null) {
-				int majorVersion = Integer.parseInt(info.getProperty(VERSION_PROPERTY).split("\\.")[0]);
-				return majorVersion >= 4;
+				String version = info.getProperty(VERSION_PROPERTY);
+				if (StringUtils.hasText(version)) {
+					int majorVersion = Integer.parseInt(version.split("\\.")[0]);
+					return majorVersion >= 4;
+				}
+				else {
+					return false;
```

Comment:
I wonder if we should fall back to a `spring.integration.properties`?

Clearly, their version supports the operation.

Reply:
Huh? What is the difference between `spring.integration.properties` or `redis_version` property?
Technically they are the same, only the problem that you abusing a system renaming such an important property.
I mean the same way you definitely can change the property in the `spring.integration.properties` file and also may break the system

Our goal here is to use faster delete on Redis, if it is not available by some reason we fall back to the regular blocking delete.
So, if you rename  `redis_version`, you just make your own life harder 😄 

That's why I'd stay with the `false` as a fallback.

Reply:
OK; makes sense; thanks.